### PR TITLE
Use a Python entry point for 'display_urdf'

### PR DESCRIPTION
**Pull request by cottsay**
On many platforms, the unversioned python executable is not Python 2. A better solution here would be to change this package to use an entry point, as shebang python scripts don't work correctly on Windows.

This same change applies to Noetic, but there isn't a Noetic-specific branch to make the change against.

### CODE REVIEW

**Review by sloretz — DISMISSED** [review]
> A better solution here would be to change this package to use an entry point, as shebang python scripts don't work correctly on Windows.

Agreed. I'd much rather see that fix get implemented: remove the shebang, make the script not executable, and add a `console_scripts` entrypoint in the `setup.py` for ROS 2.


> This same change applies to Noetic, but there isn't a Noetic-specific branch to make the change against.

I don't think any changes are needed in ROS 1. ROS Melodic and Noetic [use `catkin_install_python()` to rewrite the shebang](http://wiki.ros.org/UsingPython3/SourceCodeChanges#Changing_shebangs) in the devel space and when it is installed.

https://github.com/ros/urdf_parser_py/blob/b6a9726c976933fb8cec0b5b872101984ec131dd/CMakeLists.txt#L9-L12

**Comment by cottsay** [comment]
>  remove the shebang, make the script not executable, and add a `console_scripts` entrypoint in the `setup.py`

Done. I split it into three commits. The first resolves a ROS 2 bug, the second does the move, and the third fixes up the script as an entry point. We should rebase-merge this to preserve the change history of `display_urdf`.

> I don't think any changes are needed in ROS 1

You're right, I forgot about that. Good call - no changes necessary for ROS 1.

File: src/urdf_parser_py/display_urdf.py
Comment (clalancette) [inline src/urdf_parser_py/display_urdf.py]: Nit: in ROS 2 we typically use `args` instead of `argv` (though we aren't 100% consistent, sigh).

**Review by clalancette — APPROVED** [review]
This looks good to me; I'll leave it to you whether you want to fix the nit or not.

I'd prefer to wait to merge this until @sloretz has had another chance to look it over.

File: src/urdf_parser_py/display_urdf.py
Comment (sloretz) [reply src/urdf_parser_py/display_urdf.py]: While `args/argv` argument has become widespread, I'm pretty sure it doesn't actually work. I think scripts still have to get arguments from `sys.argv`.

The template for `console_scripts` doesn't pass anything - see the function call on the entry point is just`()`.
https://github.com/pypa/setuptools/blob/990237c64404876791dc5985e7cfd260852dac0e/setuptools/command/easy_install.py#L2069

It's also not mentioned in the docs.
https://python-packaging.readthedocs.io/en/latest/command-line-scripts.html#the-console-scripts-entry-point

File: src/urdf_parser_py/display_urdf.py
Comment (sloretz) [inline src/urdf_parser_py/display_urdf.py]: I'd be surprised if argv is ever not `None`. This happens to be ok because it looks like the first argument is `prog` which is supposed to be the name of the program, else `sys.argv[0]` if `None`. https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser

Still I think this should be removed for clarity.

**Review by sloretz — CHANGES_REQUESTED** [review]

**Comment by clalancette** [comment]
@cottsay Friendly ping here; I think this is close to getting done, with just a couple more nits to fix from @sloretz .

**Comment by cottsay** [comment]
Okay, I dropped the `argv` stuff. Again, please rebase-merge this PR when you're happy with it - I updated the commits to include a reference to this PR so they look good when rebase-merged.

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
I'm going to go ahead and merge this, thanks for the patch @cottsay

**Pull request merged**
